app/include/zmk/endpoints_types.h

```diff
@@ -6,7 +6,21 @@
 
 #pragma once
 
-enum zmk_endpoint {
-    ZMK_ENDPOINT_USB,
-    ZMK_ENDPOINT_BLE,
+/**
+ * The method by which data is sent.
+ */
+enum zmk_transport {
+    ZMK_TRANSPORT_USB,
+    ZMK_TRANSPORT_BLE,
+};
+
+/**
+ * A specific endpoint to which data may be sent.
+ */
+struct zmk_endpoint_instance {
+    enum zmk_transport transport;
+    union {
+        // ZMK_TRANSPORT_USB: no data
+        int ble_profile_index; // ZMK_TRANSPORT_BLE
```

Comment:
I wonder, since we might have someone add a custom 2.4Ghz transport later, for instance, if we would want to do something like:


```suggestion
        struct zmk_transport_ble_details ble; // ZMK_TRANSPORT_BLE
```

And

```
struct zmk_transport_ble_details {
    int profile_index;
};
```

So you'd access `endpoint.ble.profile_index` to be *really* explicit.

Thoughts?

Reply:
I used an anonymous union here so that things unlikely to have any extra data wouldn't need that extra level, but you could still add them for things that needed them. That said, there really isn't much of a difference between
```c
endpoint.ble_profile_index
```
and
```c
endpoint.ble.profile_index
```
so I think I'll go with what you suggested. It would also help avoid needing to update a lot of code if we ever needed to add a new field for an existing transport.

Reply:
Yeah, avoiding destructive refactors later is definitely an added bonus.

Reply:
Done.